enterprise/views.py

```diff
@@ -847,6 +859,47 @@ def post(self, request):
         return redirect(success_url if consent_provided else failure_url)
 
 
+class EnterpriseLoginView(FormView):
+    """
+    Allow an enterprise learner to login by enterprise customer's slug login.
+    """
+
+    form_class = EnterpriseLoginForm
+    template_name = 'enterprise/enterprise_customer_login_page.html'
+
+    def get_context_data(self, **kwargs):
+        """Return the context data needed to render the view."""
+        context_data = super(EnterpriseLoginView, self).get_context_data(**kwargs)
+        context_data.update({
+            'page_title': _(u'Enterprise Slug Login'),
+            'enterprise_login_title_message': ENTERPRISE_LOGIN_TITLE,
+            'enterprise_login_subtitle_message': ENTERPRISE_LOGIN_SUBTITLE,
+        })
+        return context_data
+
+    def form_invalid(self, form):
+        """
+        If the form is invalid then return the errors.
+        """
+        # flatten the list of lists
+        errors = [item for sublist in form.errors.values() for item in sublist]
+        return JsonResponse({'errors': errors}, status=400)
+
+    def form_valid(self, form):
+        """
+        If the form is valid, redirect to the third party auth login page.
+        """
+        provider_login_url = get_provider_login_url(
```

Comment:
`get_provider_login_url ` will be None in case of import error.
I guess you want to write something like
```
# This case will only happend when we try to run the edx-enterprise independently.
if not get_provider_login_url:
      return JsonResponse({'errors': [ERROR_MESSAGE_FOR_SLUG_LOGIN]}, status=400)
provider_login_url = get_provider_login_url(
       self.request,
       get_enterprise_customer_idp(form.cleaned_data['enterprise_slug']).provider_id
)
return JsonResponse({"url": provider_login_url})

```

Reply:
Typo. you are right.